File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Nullness - Value type 01.fs
module Module

open System
open System.Collections.Generic

let a: Nullable<int> = Class.NullableInt()
let b: List<int> = Class.Ints()
let c: KeyValuePair<string, int> = Class.Pair()
let d: Nullable<KeyValuePair<int, string>> = Class.NullablePair()

File: ReSharper.FSharp/test/data/common/assemblyReaderShim/Nullness - Value type 01.cs
#nullable enable
using System.Collections.Generic;

public class Class
{
  public static int? NullableInt() => null;
  public static List<int> Ints() => null!;
  public static KeyValuePair<string?, int> Pair() => default;
  public static KeyValuePair<int, string?>? NullablePair() => null;
}
